### PR TITLE
refactor: remove foundry-abi

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Part 2/2 of removing `foundry-abi`.

Follow-up to #6379

~~Needs https://github.com/alloy-rs/core/pull/426~~

### CODE REVIEW

File: crates/common/src/fmt/console.rs [hunk 1]
```diff
@@ -71,10 +71,13 @@ impl ConsoleFmt for U256 {
             FormatSpec::String | FormatSpec::Object | FormatSpec::Number | FormatSpec::Integer => {
                 self.pretty()
             }
-            FormatSpec::Hexadecimal => format!("0x{:x}", *self),
+            FormatSpec::Hexadecimal => {
+                let hex = format!("{self:x}");
+                format!("0x{}", hex.trim_start_matches('0'))
+            }
```
Comment (DaniPopes) [inline crates/common/src/fmt/console.rs:77]: ruint hex fmt pitfall

File: crates/common/src/fmt/console.rs [hunk 2]
```diff
@@ -315,86 +320,88 @@ mod tests {
...
-        assert_eq!(
-            "'0xdEADBEeF00000000000000000000000000000000'",
-            console_log_format_1("%o", &addr)
-        );
+        assert_eq!("0xdEADBEeF00000000000000000000000000000000", fmt_1("%s", &addr));
+        assert_eq!("NaN", fmt_1("%d", &addr));
+        assert_eq!("NaN", fmt_1("%i", &addr));
+        assert_eq!("NaN", fmt_1("%e", &addr));
+        assert_eq!("0xdEADBEeF00000000000000000000000000000000", fmt_1("%x", &addr));
+        assert_eq!("'0xdEADBEeF00000000000000000000000000000000'", fmt_1("%o", &addr));
```
Comment (DaniPopes) [inline crates/common/src/fmt/console.rs:368]: Added hex formatting for consistency with other byte arrays

File: crates/macros/Cargo.toml [hunk 3]
```diff
@@ -9,10 +9,14 @@ license.workspace = true
...
 
-[dependencies]
-foundry-macros-impl = { path = "impl" }
-
-ethers-core.workspace = true
+[lib]
+proc-macro = true
+# proc-macro tests aren't fully supported by cargo-nextest archives
+test = false
+doc = false
```
Comment (DaniPopes) [inline crates/macros/Cargo.toml:16]: moved all non-macro stuff to common, so flattened macros-impl into one crate

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**